src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -366,6 +369,29 @@ public async Task<SyncTriggersPayload> GetSyncTriggersPayload()
             };
         }
 
+        internal JObject GetHostJson()
+        {
+            var defaultJObject = new JObject();
+            var scriptPath = _applicationHostOptions.CurrentValue?.ScriptPath;
+            string hostFilePath = Path.Combine(scriptPath, ScriptConstants.HostMetadataFileName);
+            try
+            {
+                _logger.LogInformation("Reading host.json for SyncTrigger.");
+                string json = File.ReadAllText(scriptPath);
+                return JObject.Parse(json);
+            }
+            catch (JsonException ex)
+            {
+                _logger.LogWarning($"Unable to parse host configuration file '{scriptPath}'. : {ex.Message} : {ex.StackTrace}");
+                return defaultJObject;
+            }
+            catch (Exception ex) when (ex is FileNotFoundException || ex is DirectoryNotFoundException)
+            {
+                _logger.LogWarning($"Unable to find host configuration file '{scriptPath}'. : {ex.Message} : {ex.StackTrace}");
```

Comment:
Why do you separate `ex.Message` from `ex.StackTrace`? Isn't it better to just trace the full exception details?
```suggestion
                _logger.LogWarning($"Unable to find host configuration file '{scriptPath}'. : {ex}");
```